### PR TITLE
Added a flag for enabling the application of the TED and TRE tables.

### PR DESCRIPTION
**Pull request by aehart**
#### PR description:

This is a port to CMSSW_11_2_X of #30818, which disables the lookup tables in the TrackletEngineDisplaced and TripletEngine, which were found to cause excessive memory consumption.

### CODE REVIEW

**Comment by silviodonato** [comment]
@aehart  please apply the code-check fixes and propagate them to https://github.com/cms-sw/cmssw/pull/30818 too. Thanks

**Comment by silviodonato** [comment]
please test

**Comment by silviodonato** [comment]
please test
@rekovic

File: L1Trigger/TrackFindingTracklet/interface/Settings.h [hunk 1]
```diff
@@ -646,10 +650,16 @@ namespace trklet {
 
     unsigned int writememsect_{3};  //writemem only for this sector (note that the files will have _4 extension)
 
-    bool writeTripletTables_{false};  //Train and write the TED and TRE tables. N.B.: the tables
-                                      //cannot be applied while they are being trained, i.e.,
-                                      //this flag effectively turns off the cuts in
-                                      //TrackletEngineDisplaced and TripletEngine
+    //FIXME: The TED and TRE tables are currently under development. They use
```
Comment (rekovic) [inline L1Trigger/TrackFindingTracklet/interface/Settings.h:653]: What does FIXME mean?  This PR is going to be followed by another PR on this code?
If not, and you are suggesting that TED and TRE tables might/will undergo changes (content/format) please remove this comment as it introduces confusion.

**Comment by rekovic** [comment]
otherwise fine

File: L1Trigger/TrackFindingTracklet/interface/Settings.h (hunk 1 above)
Comment (aehart) [reply L1Trigger/TrackFindingTracklet/interface/Settings.h]: There likely will be changes to the code as we continue to develop the tables, but not in the immediate future. So I removed the comment.

**Comment by silviodonato** [comment]
please test

**Comment by silviodonato** [comment]
kind reminder @rekovic

**Comment by silviodonato** [comment]
kind reminder @cms-sw/l1-l2

**Comment by rekovic** [comment]
pending decision on #30818

**Comment by silviodonato** [comment]
I'm converting this PR as a draft just to remove this from the ORP spreadsheet. Of course feel free to open it back if needed.

**Comment by silviodonato** [comment]
@rekovic @aehart @skinnari @tomalin I missed your comments in https://github.com/cms-sw/cmssw/pull/30818 because I was on holiday. As soon as we managed to start the HLT TDR re-reco production, we decided to keep everything unchanged to avoid any possible risk.

I think it would be good to have it in master. If you want to reactivate this PR, @aehart please push on the button "Ready for review"
